cpp/src/io/parquet/reader_impl_helpers.hpp

```diff
@@ -237,6 +242,49 @@ class aggregate_reader_metadata {
     host_span<std::vector<size_type> const> row_group_indices,
     host_span<int const> column_schemas) const;
 
+  /**
+   * @brief Filters the row groups using stats filter
+   *
+   * @param input_row_group_indices Lists of input row groups, one per source
+   * @param total_row_groups Total number of row groups in `input_row_group_indices`
+   * @param output_dtypes Datatypes of output columns
+   * @param output_column_schemas schema indices of output columns
+   * @param filter AST expression to filter row groups based on bloom filter membership
+   * @param stream CUDA stream used for device memory operations and kernel launches
+   *
+   * @return Filtered row group indices if any is filtered
+   */
+  [[nodiscard]] std::optional<std::vector<std::vector<size_type>>> apply_stats_filters(
+    host_span<std::vector<size_type> const> input_row_group_indices,
+    size_type total_row_groups,
+    host_span<data_type const> output_dtypes,
+    host_span<int const> output_column_schemas,
+    std::reference_wrapper<ast::expression const> filter,
+    rmm::cuda_stream_view stream) const;
+
+  /**
+   * @brief Filters the row groups using bloom filters
+   *
+   * @param bloom_filter_data Bloom filter data device buffers for each input row group
+   * @param input_row_group_indices Lists of input row groups, one per source
+   * @param total_row_groups Total number of row groups in `input_row_group_indices`
+   * @param output_dtypes Datatypes of output columns
+   * @param equality_col_schemas schema indices of equality columns only
+   * @param filter AST expression to filter row groups based on bloom filter membership
+   * @param stream CUDA stream used for device memory operations and kernel launches
+   *
+   * @return A pair of filtered row group indices if any is filtered.
```

Comment:
pair?

Reply:
Thanks for catching this. The return type is not a pair (row groups, bool) anymore as we only call this if and only if bloom filter is available.

Reply:
Fixed now!